### DISCUSSION TITLE
[Feature] Encrypted Backup to Untrusted Servers

### DISCUSSION THREAD

**Discussion by jflattery** [immich-app/immich#10786] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

As an Immich user, I want to securely back up my media files (images, videos) and relevant metadata to servers that I do not fully trust or control. This includes the ability to use both a primary trusted server and a secondary untrusted server, ensuring that my data remains private and secure through encryption. Thinking of the Bitwarden model.

**Description:**
In certain scenarios, users may need to back up their data to servers that they do not have full access to or control over. In these cases, users may not fully trust the security of these servers. To address this concern, I propose the following features:

1. **Encrypted Backup to Untrusted Servers:**
   - Allow users to back up their media files and metadata as encrypted blobs to untrusted servers.
   - Ensure that neither the server nor the owner of the server can access the content of the backups.
   - Implement user-side decryption, allowing users to view their content within the Immich application and web application, similar to Bitwarden's approach.

2. **Server Configuration Options:**
   - Provide users with the ability to configure both a primary trusted server and a secondary untrusted server.
   - Allow the primary server to handle server-side activities such as deduplication, object detection, and face recognition.
   - Enable the primary server to sync encrypted data to the secondary untrusted server.

3. **User Choices for Server Usage:**
   - Offer users the choice to:
     - Use a single trusted server.
     - Use a single untrusted server with encrypted backups.
     - Utilize both a primary trusted server and a secondary untrusted server for encrypted backups.

**Acceptance Criteria:**
1. Users can configure encrypted backups to untrusted servers.
2. Encrypted blobs are stored on the untrusted server, ensuring data privacy.
3. Users can view their encrypted content in the Immich application and web application using user-side decryption.
4. Users can configure both a primary trusted server and a secondary untrusted server.
5. The primary server can perform server-side activities and sync encrypted data to the secondary server.
6. Users can select their preferred server configuration option from the provided choices.

**Additional Notes:**
- Users acknowledge that using an untrusted server will result in the loss of certain server-side features (e.g., deduplication, object detection, face recognition).
- The implementation should prioritize user privacy and security, ensuring that encryption and decryption are handled securely on the client side.

Tagging @chrislas as he expressed an interest in this in the Selfhosted podcast.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Duplicate of #450
